src/internal/storage/chunk/reader.go

```diff
@@ -60,9 +66,36 @@ func (r *Reader) Iterate(cb func(*DataReader) error) error {
 }
 
 // Get writes the concatenation of the data referenced by the data references.
-func (r *Reader) Get(w io.Writer) error {
+func (r *Reader) Get(w io.Writer) (retErr error) {
+	if len(r.dataRefs) <= 1 {
+		return r.Iterate(func(dr *DataReader) error {
+			return dr.Get(w)
+		})
+	}
+	ctx, cancel := context.WithCancel(r.ctx)
+	taskChain := NewTaskChain(ctx)
+	defer func() {
+		if retErr != nil {
+			cancel()
+		}
+		if err := taskChain.Wait(); retErr == nil {
+			retErr = err
+		}
+	}()
+	sem := semaphore.NewWeighted(int64(r.prefetchLimit))
 	return r.Iterate(func(dr *DataReader) error {
-		return dr.Get(w)
+		if err := sem.Acquire(ctx, 1); err != nil {
```

Comment:
Why don't we Acquire inside CreateTask?

Reply:
The goroutine would exist at that point, we want to acquire before that.